docs/audio.rst

```diff
@@ -215,16 +225,17 @@ Sound Effects Example
 .. include:: ../examples/soundeffects.py
     :code: python
 
+
 AudioFrame
 ==========
 
 .. py:class::
-    AudioFrame
+    AudioFrame(size=32)
```

Comment:
Perhaps this could be extended to take optional keyword arguments `duration` and `rate` (with a default), to make it easy to create an `AudioFrame` of a given duration.

Reply:
I think in this case, similar to the previous comment about having an `AudioFrame.set_rate()`, if `rate` was added to the constructor then we'd need to have `set_rate()` as a method changing each instance value, instead of a class static function.

I do like the idea to be able to make the arguments of the `AudioFrame` constructor simpler to work out if the user just wants a specific "size in time".

What would happen if the constructor is provided an argument for `size` and for `duration`? Would that throw an exception?

Reply:
Note that there's a difference between record and play rates.  The rate here is related to the recording rate, although the actual recording rate is set when you call `microphone.record_into()`.  And that's different again to the playback rate set by `.set_rate()`.



> What would happen if the constructor is provided an argument for `size` and for `duration`? Would that throw an exception?

Yes it could throw an exception.

Reply:
Discussion about specifying the size of an `AudioFrame`:
- https://github.com/microbit-foundation/micropython-microbit-v2/issues/193